#168095164 Fix password reset link sent to user email

kennyedward opened the pull request:
#### What does this PR do?
This pull request updates the reset link sent to a user email

#### Description of Task to be completed
- Reset and update user password

#### How should this be manually tested?
- Clone and cd into this repository
- run ```npm run start:dev```
- Using _Postman_ make a **POST** request with ```{ email: someone@gmail.com }``` to http://localhost:{Port}/api/v1/auth/reset_password/ to get the reset link
- Then make a **PATCH** request to http://localhost:{Port}/api/v1/auth/update_password/:id/:token
to update the password

#### Any Background Context You want to provide?
The user requesting password reset/update must be a  registered user

#### What are the relevant pivotal tracker stories?

[#168095164](https://pivotaltracker.com/story/show/168095164)

#### Screenshots (if appropriate)
<img width="1680" alt="Reste Link" src="https://user-images.githubusercontent.com/18053279/63648466-0fd8da80-c728-11e9-89d2-9bc4ad591b87.png">
<img width="1680" alt="Password Update Success" src="https://user-images.githubusercontent.com/18053279/63648473-1b2c0600-c728-11e9-9c74-ecaa42817e5a.png">

#### Questions:
N/A

successgilli commented on tests/routes/bookmarks.test.js:
```diff
@@ -126,4 +114,33 @@ describe('Bookmark Endpoint', () => {
+        .get(`${baseRoute}/articles/bookmarks`)
+        .set('Authorization', `Bearer ${userToken}`);
+
+      response.should.have.status(200);
```
Nice job adding more tests.

successgilli commented on src/services/notification.js:
```diff
@@ -320,6 +320,20 @@ class Notification extends EventEmitter {
+   * @function
+   *
+   * @param {Array} payload - An Object that contains the user email
+   *
```
Please, why is there more than one line space before the return definition in the jsdocs of this function?

kennyedward replied:
Thanks for catching that. It is fixed now.

successgilli reviewed (commented):
Nice work @kennyedward 
- Please, change the scope of the commit message 'reset password' according to the wiki.
If you must use two words, concatenate them with a hyphen.
- Change the title of the PR to conform to standards on the wiki.

kennyedward commented:
> Nice work @kennyedward
> 
> * Please, change the scope of the commit message 'reset password' according to the wiki.
>   If you must use two words, concatenate them with a hyphen.
> * Change the title of the PR to conform to standards on the wiki.

Thanks for the feedback. However, in the team convention doc, I couldn't locate where it was stated but I found **"It should also be a maximum of 2 words."**
For the PR - It's updated.

successgilli commented:
> > Nice work @kennyedward
> > 
> > * Please, change the scope of the commit message 'reset password' according to the wiki.
> >   If you must use two words, concatenate them with a hyphen.
> > * Change the title of the PR to conform to standards on the wiki.
> 
> Thanks for the feedback. However, in the team convention doc, I couldn't locate where it was stated but I found **"It should also be a maximum of 2 words."**
> For the PR - It's updated.

successgilli commented:
> > Nice work @kennyedward
> > 
> > * Please, change the scope of the commit message 'reset password' according to the wiki.
> >   If you must use two words, concatenate them with a hyphen.
> > * Change the title of the PR to conform to standards on the wiki.
> 
> Thanks for the feedback. However, in the team convention doc, I couldn't locate where it was stated but I found **"It should also be a maximum of 2 words."**
> For the PR - It's updated.

successgilli commented:
> > Nice work @kennyedward
> > 
> > * Please, change the scope of the commit message 'reset password' according to the wiki.
> >   If you must use two words, concatenate them with a hyphen.
> > * Change the title of the PR to conform to standards on the wiki.
> 
> Thanks for the feedback. However, in the team convention doc, I couldn't locate where it was stated but I found **"It should also be a maximum of 2 words."**
> For the PR - It's updated.

Sorry for closing the PR initially.
I wanted to find an option to refer you to a comment @iverenshaguy  made on @Bastieno's most recent PR, then I hit close.
![image](https://user-images.githubusercontent.com/42284402/63649061-cf318f00-c730-11e9-91ed-945deabcdac4.png)

kennyedward committed:
fix(password-update): update password reset link

- update password reset link sent to user email
- update .env and app.json file
- add tests to bookmark endpoint

[Finishes #168095164]

iverenshaguy commented:
> > Nice work @kennyedward
> > 
> > * Please, change the scope of the commit message 'reset password' according to the wiki.
> >   If you must use two words, concatenate them with a hyphen.
> > * Change the title of the PR to conform to standards on the wiki.
> 
> Thanks for the feedback. However, in the team convention doc, I couldn't locate where it was stated but I found **"It should also be a maximum of 2 words."**
> For the PR - It's updated.

Actually @kennyedward, 2 words concatenated with a dash. Please update this in the wiki. @successgilli thanks for catching this.

kennyedward commented:
> > > Nice work @kennyedward
> > > 
> > > * Please, change the scope of the commit message 'reset password' according to the wiki.
> > >   If you must use two words, concatenate them with a hyphen.
> > > * Change the title of the PR to conform to standards on the wiki.
> > 
> > 
> > Thanks for the feedback. However, in the team convention doc, I couldn't locate where it was stated but I found **"It should also be a maximum of 2 words."**
> > For the PR - It's updated.
> 
> Actually @kennyedward, 2 words concatenated with a dash. Please update this in the wiki. @successgilli thanks for catching this.

It is now updated. Thanks, @iverenshaguy @successgilli.

iverenshaguy reviewed (approved):
LGTM
